src/polyfem/assembler/GenericProblem.cpp

```diff
@@ -80,6 +86,130 @@ namespace polyfem
 				form_ = 0;
 		}
 
+		double PiecewiseConstantInterpolation::eval(const double t) const
+		{
+			if (t < points_[0])
+				return 0;
+			for (size_t i = 0; i < points_.size()-1; ++i) {
+				if (t >= points_[i] && t < points_[i+1]) {
+					return values_[i];
+				}
+			}
+			if (extend_ == 0)
+				return values_[values_.size()-1];
+			else if (extend_ == 1)
+				return values_[values_.size()-1]*t;
+			else if (extend_ == 2) {
+				for (size_t i = 0; i < points_.size()-1; ++i) {
+					if (t-points_[points_.size()-1] >= points_[i] && t-points_[points_.size()-1] < points_[i+1]) {
```

Comment:
For this just make it a (recursive) call to `eval(fmod(t, points[-1]) + points[0])+offset`.